### PR TITLE
Viewing Released Peer Reviews 

### PR DESCRIPTION
**Pull request by christinem**
This PR implements the views for both Students and Admins to view released Peer Reviews. The admin peer review submissions table is not correct UI-wise yet, but the "Group Name" link properly does link to the reviews that said reviewer has made.

### CODE REVIEW

File: app/controllers/peer_reviews_controller.rb [hunk 1]
```diff
@@ -4,7 +4,8 @@ class PeerReviewsController < ApplicationController
   include RandomAssignHelper
 
   before_action :set_peer_review, only: [:show, :edit, :update, :destroy]
-  before_filter :authorize_only_for_admin
+
+  before_filter :authorize_only_for_admin, except: [:show_reviews, :show_result]
```
Comment (david-yz-liu) [inline app/controllers/peer_reviews_controller.rb]: Need to add a `authorize_for_user` for the new actions

File: app/controllers/peer_reviews_controller.rb [hunk 2]
```diff
@@ -32,6 +33,31 @@ def populate
...
   end
 
+  def show_reviews
+    assignment = Assignment.find(params[:assignment_id])
+    # grab the first peer review of the reviewee group
+    pr = @current_user.grouping_for(assignment.id).peer_reviews.first
+
+    if !pr.nil?
+      redirect_to show_result_assignment_peer_review_path(assignment.id, id: pr.id)
+
```
Comment (david-yz-liu) [inline app/controllers/peer_reviews_controller.rb]: Remove this line.

File: app/controllers/peer_reviews_controller.rb [hunk 3]
```diff
@@ -32,6 +33,31 @@ def populate
...
+    else
+      render 'shared/http_status', formats: [:html],
+             locals: { code: '404',
+                       message: HttpStatusHelper::ERROR_CODE[
+                           'message']['404'] }, status: 404,
+             layout: false
+      return
```
Comment (david-yz-liu) [inline app/controllers/peer_reviews_controller.rb]: Don't need the return

File: app/controllers/peer_reviews_controller.rb [hunk 4]
```diff
@@ -32,6 +33,31 @@ def populate
...
+    end
+
```
Comment (david-yz-liu) [inline app/controllers/peer_reviews_controller.rb]: Don't put blank lines between `end`

File: app/controllers/results_controller.rb [hunk 5]
```diff
@@ -444,26 +443,55 @@ def update_mark
   end
 
   def view_marks
+    #debugger
```
Comment (david-yz-liu) [inline app/controllers/results_controller.rb]: Remove this.

File: app/controllers/results_controller.rb [hunk 6]
```diff
@@ -444,26 +443,55 @@ def update_mark
...
     @assignment = Assignment.find(params[:assignment_id])
-    @grouping = current_user.accepted_grouping_for(@assignment.id)
+
+    if !current_user.student?
```
Comment (david-yz-liu) [inline app/controllers/results_controller.rb]: Reverse the branches.

File: app/controllers/results_controller.rb [hunk 7]
```diff
@@ -491,7 +519,18 @@ def view_marks
...
     @marks_map = Hash.new
     @old_marks_map = Hash.new
-    @mark_criteria = @assignment.get_criteria(:ta)
+
+    if @result.is_a_review?
+      if @current_user.is_reviewer_for?(@assignment.pr_assignment, @result) ||
+        @first_file.belongs_to?(@current_user)
+        @mark_criteria = @assignment.get_criteria(:peer)
+      else
+        @mark_criteria = @assignment.pr_assignment.get_criteria(:ta)
```
Comment (david-yz-liu) [inline app/controllers/results_controller.rb]: The peer criteria should always be shown.

File: app/helpers/submissions_helper.rb [hunk 8]
```diff
@@ -59,6 +59,18 @@ def get_submissions_table_info(assignment, groupings)
           # Get the respective reviewee's result from grouping
           result_pr = current_user.grouping_for(assignment.id).review_for(grouping)
           result = Result.find(result_pr.result_id)
+
+        elsif (assignment.is_peer_review? and !current_user.student?)
```
Comment (david-yz-liu) [inline app/helpers/submissions_helper.rb]: No need for parentheses.

Use `&&` not `and`

File: app/controllers/results_controller.rb [hunk 9]
```diff
@@ -384,12 +385,10 @@ def codeviewer
...
     if current_user.student?
-      # If result is a review and user doesn't have membership status for the grouping
-      # this file belongs to, or if result is a review and the student is not a reviewer of this
-      # result, then student does not have access to this file. Display an error.
-
-      if (!@result.is_a_review? && @file.submission.grouping.membership_status(current_user).nil?) ||
-          (@result.is_a_review? && !current_user.is_reviewer_for?(@assignment.pr_assignment, @result))
+      # Unless this file belongs to this user or this user is a reviewer of this result,
+      # this student isn't authorized to view these files. Display an error
+      unless @file.belongs_to?(current_user) ||
```
Comment (david-yz-liu) [inline app/controllers/results_controller.rb]: Don't make an instance method of SubmissionFile. Check a different way. You have access to the `@grouping`.

File: app/views/results/common/_peer_reviews.html.erb [hunk 10]
```diff
@@ -0,0 +1,32 @@
+<div id="peer_reviews">
```
Comment (david-yz-liu) [inline app/views/results/common/_peer_reviews.html.erb]: Use single quotes.

File: app/views/results/common/_peer_reviews.html.erb [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  <p id='peer_review_selector_label'>
+    <%= @current_user.student? ? "Reviews made on your submission: " : "This reviewer's reviews: " %>
```
Comment (david-yz-liu) [inline app/views/results/common/_peer_reviews.html.erb]: These strings need to be internationalized.

File: app/views/results/common/_peer_reviews.html.erb [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  <div id='peer_review_selector' class='dropdown'>
+    <a><%= "Review " + (prs.index(current_pr) + 1).to_s %></a>
+    <ul>
+      <% prs.each_with_index do |pr, i| %>
+        <% if @current_user.student? %>
+          <li> <%= link_to("Review " + (i + 1).to_s,
+                        view_marks_assignment_result_path(assignment.id, pr.result_id)) %> </li>
+        <% else %>
+            <li> <%= link_to("Review " + (i + 1).to_s,
+                             view_marks_assignment_submission_result_path( assignment_id: assignment.id,
```
Comment (david-yz-liu) [inline app/views/results/common/_peer_reviews.html.erb]: Remove the extraneous space

File: app/views/results/marker/_marker_rubric_criterion_li.html.erb [hunk 13]
```diff
@@ -32,12 +32,15 @@
 
               if level_desc && !level_desc.empty? -%>
                 <tr>
-                  <td id='mark_<%= mark.id %>_<%= num_levels %>' class='<%= td_class %>'
+                  <td id='mark_<%=mark.id%>_<%=num_levels%>' class='<%=td_class%>'
```
Comment (david-yz-liu) [inline app/views/results/marker/_marker_rubric_criterion_li.html.erb]: Revert this change. It happens at least once below, as well.

File: app/views/results/student/_student_summary.html.erb [hunk 14]
```diff
@@ -77,50 +77,68 @@
...
-      <span><%= result.get_subtotal %> / <%= assignment.max_mark %></span>
-  </div>
+  <% unless @result.is_a_review? %>
 
-  <div class='submission_rule_pane'>
-    <div class='submission_rule_label'>
-      <strong><%= t('marker.marks.submission_rule') %>:</strong>
+    <div class='summary_block'>
+        <strong><%= t('marker.marks.subtotal') %></strong>
+        <span><%= result.get_subtotal %> / <%= assignment.max_mark(@result.is_a_review? ? :peer : :ta) %></span>
```
Comment (david-yz-liu) [inline app/views/results/student/_student_summary.html.erb]: This repeats a check that already happened.

**Comment by david-yz-liu** [comment]
@christinem several comments. I would have liked to see this pull request much sooner.

There are several strings which need to be internationalized that I did not point out.

**Comment by david-yz-liu** [comment]
@christinem you can do another pull from master. @GitGeminiano fixed the failing tests there.

**Comment by christinem** [comment]
@david-yz-liu Thanks David, I've now pulled master into this branch.
